app/models/user.rb

```diff
@@ -2,6 +2,11 @@
 
 class User < ApplicationRecord
   attr_accessor :grant_permission_role
+  attr_accessor :pending_reviewer_invitations
+
+  before_create :set_pending_reviewer_invitations, prepend: true
+  after_commit  :process_pending_reviewer_invitations,  on: :create,
```

Comment:
I think after_commit is not necessary in this case. after_commit is suitable for interacting with external system that can't be rolled back with in the database transaction. But it seems that processing pending invitation only involves database.

Reply:
i think `after_create` failed in an earlier iteration but then `after_commit` didn't cause a problem moving forward.